include/aws/s3/s3_client.h

```diff
@@ -481,6 +483,15 @@ struct aws_s3_meta_request_options {
     /* The type of meta request we will be trying to accelerate. */
     enum aws_s3_meta_request_type type;
 
+    /**
+     * Optional.
+     * The S3 operation name (e.g. "CreateBucket").
+     * You should only set this when type is AWS_S3_META_REQUEST_TYPE_DEFAULT;
+     * it is automatically populated for other meta-request types.
+     * This name is used to fill out details in metrics and error reports.
+     */
+    struct aws_byte_cursor operation_name;
```

Comment:
I think we're overcomplicating things by using both an enum and a string for the same data. It seems unnecessary to have a request-level variable set only in one specific case just to return it later. If a user sets 'default', returning NULL or 'Default' should be sufficient. User can include any additional information in the user_data of the request.

Reply:
In most cases, it is just repeated information, like type = getObject, operationName = getObject.

Reply:
I eliminated a lot of the repeated typing.
But I still think it's worth having both enum and string, so we can let users pass through the names of the N different S3 operations we don't have explicit enums for.